sonic-psud/scripts/psud

```diff
@@ -133,8 +149,111 @@ def log_on_status_changed(logger, normal_status, normal_log, abnormal_log):
     else:
         logger.log_warning(abnormal_log)
 
-
 #
+# PSU Chassis Info ==========================================================
+#
+class PsuChassisInfo(logger.Logger):
+
+    def __init__(self, log_identifier, chassis):
+        """
+        Constructor for PsuChassisInfo
+        :param chassis: Object representing a platform chassis
+        """
+        super(PsuChassisInfo, self).__init__(log_identifier)
+
+        self.chassis = chassis
+        self.master_status_good = True
+        self.total_consumed_power = 0.0
+        self.total_supplied_power = 0.0
+
+    def run_power_budget(self, chassis_tbl):
+        self.total_supplied_power = 0.0
+        self.total_consumed_power = 0.0
+        total_supplied_power = 0.0
+        total_fan_consumed_power = 0.0
+        total_module_consumed_power = 0.0
+
+        dict_index = 0
+        total_entries_len = 2 #For total supplied and consumed
+        dict_len = self.chassis.get_num_psus() +\
+                   self.chassis.get_num_fan_drawers() +\
+                   self.chassis.get_num_modules() + \
+                   total_entries_len
+
+        fvs = swsscommon.FieldValuePairs(dict_len)
+
+        for index, psu in enumerate(self.chassis.get_all_psus()):
```

Comment:
Assuming these API's psu.get_maximum_supplied_power, power_consumer.get_maximum_consumed_power etc are from PSUd running in supervisor card, talking to Redis DB's in the linecards... Will this operation be a bit heavy to be done every 3 secs ?  Any observations 

Can each module calculate it's own  total_supplied_power, total_consumed_power -- and the supervisor card just do calculations with that data ?

Reply:
It would not be querying the redis-DB on the line-card. Instead, vendor's implementation will return constant values based on the card-type (line-card or SFM) or fan-type (version1, version 2 etc). So, the APIs should not doing rpc calls.